prow/config/jobs.go

```diff
@@ -171,6 +171,12 @@ type Presubmit struct {
 	// (Default: `/test <job name>`)
 	RerunCommand string `json:"rerun_command,omitempty"`
 
+	// AlwaysRunByTide indicates that a job should always run by Tide as long as
+	// Brancher matches.
+	// This is used when a prowjob is so expensive that it's not ideal to run on
+	// every single push from all PRs.
+	AlwaysRunByTide bool `json:"always_run_by_tide"`
```

Comment:
Nit: rename to `RunsAfterOtherJobs` (or similar)? I think this better matches the intent in the PR description.

Reply:
or `OnlyRunOnMergeAttempts`?

Reply:
I like `OnlyRunOnMergeAttempts`, it's almost there, the only nit is the `Always` part, note that these jobs might miss triggers and they are required to "Always" run by Tide, so it might be `AlwaysAndOnlyRunOnMergeAttempts`, this is more correct but also not the possibly the best name for this field. Not sure which way is better, might just stick with the current name, at least it has the "Always" part

Reply:
only is misleading if /test exists

Reply:
We have run_if_changed maybe run_before_merge

Reply:
`must_run_before_merge`?

Reply:
FWIW I like `run_before_merge` better than `must_run_before_merge`.